tokio/src/task/mod.rs

```diff
@@ -1,5 +1,202 @@
 //! Asynchronous green-threads.
-
+//!
+//! ## What are Tasks?
+//!
+//! A _task_ is a light weight, non-blocking unit of execution. A task is similar
+//! to an OS thread, but rather than being managed by the OS scheduler, they are
+//! managed by the [Tokio runtime][rt]. Key points about tasks include:
+//!
+//! * Tasks are **light weight**. Because tasks are scheduled by the Tokio
+//!   runtime rather than the operating system, creating new tasks or switching
+//!   between tasks does not require a context switch and has fairly low
+//!   overhead. Creating, running, and destroying large numbers of tasks is
+//!   quite cheap, especially compared to OS threads.
+//!
+//! * Tasks are scheduled **cooperatively**. Most operating systems implement
+//!   _preemptive multitasking_. This is a scheduling technique where the
+//!   operating system allows each thread to run for a period of time, and then
+//!   _preempts_ it, temporarily pausing that thread and switching to another.
+//!   Tasks, on the other hand, implement _cooperative multitasking_. In
+//!   cooperative multitasking, a task is allowed to run until it _yields_,
+//!   indicating to the Tokio runtime's scheduler that it cannopt currently
```

Comment:
```suggestion
//!   indicating to the Tokio runtime's scheduler that it cannot currently
```

Reply:
thanks for catching these! fixed in 1873017b0aaf5c2b55524b36822c5f68e0158642